diff --git a/media/lua/client/OptionScreens/ConnectToServer.lua b/media/lua/client/OptionScreens/ConnectToServer.lua
index 783fdf6..015b816 100644
--- a/media/lua/client/OptionScreens/ConnectToServer.lua
+++ b/media/lua/client/OptionScreens/ConnectToServer.lua
@@ -171,7 +171,7 @@ function ConnectToServer:connect(previousScreen, serverName, userName, password,
 	self.serverName:setVisible(true)
 	if serverName == "" then
 		self.serverName1:setName(getText("UI_ConnectToServer_ServerIP"))
-		self.serverName:setName(IP)
+		self.serverName:setName("***")
 	else
 		self.serverName1:setName(getText("UI_ConnectToServer_ServerName"))
 		self.serverName:setName(serverName)
diff --git a/media/lua/client/OptionScreens/ServerConnectPopup.lua b/media/lua/client/OptionScreens/ServerConnectPopup.lua
index aab1473..5ed6774 100644
--- a/media/lua/client/OptionScreens/ServerConnectPopup.lua
+++ b/media/lua/client/OptionScreens/ServerConnectPopup.lua
@@ -165,6 +165,7 @@ function ServerConnectPopup:setServer(ip, port, passwordStr)
     if not type(port) == "string" then error "port must be a string" end
     self.ip = ip;
     self.port = port;
+    self.connectTypeEntry.selected[1] = false;
 
     -- Get the most-recently edited account for this server and fill in the fields.
     -- Assumes the most-recently edited account comes last.
@@ -188,7 +189,7 @@ end
 function ServerConnectPopup:prerender()
     ISPanelJoypad.prerender(self);
     self:drawTextCentre(getText("UI_ServerConnectPopup_Label"), self.width / 2, 20, 1, 1, 1, 1, UIFont.Large);
-    self:drawTextCentre(self.ip.." : "..self.port, self.width / 2, 40, 1, 1, 1, 1, UIFont.Large);
+    self:drawTextCentre("***", self.width / 2, 40, 1, 1, 1, 1, UIFont.Large);
     self:checkFields()
 
     if self.joyfocus and self.joypadIndexY > 0 then
