diff --git a/media/lua/client/OptionScreens/ConnectToServer.lua b/media/lua/client/OptionScreens/ConnectToServer.lua
index 783fdf6..015b816 100644
--- a/media/lua/client/OptionScreens/ConnectToServer.lua
+++ b/media/lua/client/OptionScreens/ConnectToServer.lua
@@ -171,7 +171,7 @@ function ConnectToServer:connect(previousScreen, serverName, userName, password,
 	self.serverName:setVisible(true)
 	if serverName == "" then
 		self.serverName1:setName(getText("UI_ConnectToServer_ServerIP"))
-		self.serverName:setName(IP)
+		self.serverName:setName("***")
 	else
 		self.serverName1:setName(getText("UI_ConnectToServer_ServerName"))
 		self.serverName:setName(serverName)
diff --git a/media/lua/client/OptionScreens/ServerConnectPopup.lua b/media/lua/client/OptionScreens/ServerConnectPopup.lua
index aab1473..dfca179 100644
--- a/media/lua/client/OptionScreens/ServerConnectPopup.lua
+++ b/media/lua/client/OptionScreens/ServerConnectPopup.lua
@@ -188,7 +188,7 @@ end
 function ServerConnectPopup:prerender()
     ISPanelJoypad.prerender(self);
     self:drawTextCentre(getText("UI_ServerConnectPopup_Label"), self.width / 2, 20, 1, 1, 1, 1, UIFont.Large);
-    self:drawTextCentre(self.ip.." : "..self.port, self.width / 2, 40, 1, 1, 1, 1, UIFont.Large);
+    self:drawTextCentre("***", self.width / 2, 40, 1, 1, 1, 1, UIFont.Large);
     self:checkFields()
 
     if self.joyfocus and self.joypadIndexY > 0 then
@@ -220,7 +220,10 @@ function ServerConnectPopup:onOptionMouseDown(button, x, y)
     if button.internal == "CONNECT" then
         if ServerConnectPopup.instance:checkFields() then
             local localIP = ""
-            local useSteamRelay = getSteamModeActive() and self.connectTypeEntry.selected[1]
+            local useSteamRelay = false
+            if getSteamModeActive() and self.connectTypeEntry.selected[1] then
+                useSteamRelay = true
+            end
             ConnectToServer.instance:connect(self, "", self.usernameEntry:getText(), self.passwordEntry:getInternalText(),
                 self.ip, localIP, self.port, self.serverPasswordEntry:getInternalText(), useSteamRelay);
         end
